product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/ftp/FTPClientWrapper.java

```diff
@@ -189,15 +190,90 @@ public void ftpFileDownload(String sourceFilePath, File targetFilePath) throws I
         return Arrays.asList(fileDirectoryList);
     }
 
+    /**
+     * Function to create new directories in an FTP server
+     *
+     * @param dirPath directory path
+     * @throws IOException if changing directory or creating new directory fails
+     */
+    public void makeDirectories(String dirPath) throws IOException {
+        dirPath = dirPath.startsWith("/") ? dirPath.substring(1) : dirPath;
+        String[] pathElements = dirPath.split("/");
+        if (pathElements.length > 0) {
+            for (String singleDir : pathElements) {
+                boolean dirExisted = ftpClient.changeWorkingDirectory(singleDir);
+                if (!dirExisted) {
+                    boolean dirCreated = ftpClient.makeDirectory(singleDir);
+                    if (dirCreated) {
+                        log.debug("Created new directory : " + singleDir);
+                        ftpClient.changeWorkingDirectory(singleDir);
+                    } else {
+                        log.error("Could not create directory : " + singleDir);
+                        break;
+                    }
+                }
+            }
+            ftpClient.changeWorkingDirectory("/");
+        }
+    }
+
+    /**
+     * Function to check whether a directory exists in the file system in an FTP server
+     *
+     * @param dirPath absolute directory path
+     * @return boolean whether directory exists in file system
+     * @throws IOException if changing working directory fails
+     */
+    public boolean checkDirectoryExists(String dirPath) throws IOException {
+        ftpClient.changeWorkingDirectory(dirPath);
+        int returnCode = ftpClient.getReplyCode();
+        if (returnCode == 550) {
```

Comment:
Shall we make this a class level variable and refer?

Reply:
This is fixed by [https://github.com/wso2/product-ei/pull/3471](https://github.com/wso2/product-ei/pull/3471)